src/js/components/locations-configuration/LocationAddress.jsx

```diff
@@ -99,14 +100,32 @@ class LocationAddress extends Component {
   }
 
   saveAddressOfLocation(values, callback) {
-    const valuesAsAddressObject = {
-      address: {
-        ...values,
-      },
-    };
     this.props.showSpinner();
-    const locationUrl = `/api/locations/${this.state.locationId}`;
-    apiClient.post(locationUrl, flattenRequest(valuesAsAddressObject))
+    const payload = Object.keys(values).reduce((acc, curr) => {
+      // If field was edited it is undefined and we need to send empty string
+      // to remove it from location. If it is null, there is no need to send it, because
+      // it will be null be default
+      if (values[curr] === null) {
```

Comment:
Perhaps this is something worth moving to a util API client function? Like with have `parseResponse` or `flattenRequest`.